packages/x-internal-gestures/src/core/Gesture.ts

```diff
@@ -294,6 +320,13 @@ export abstract class Gesture<GestureName extends string> {
    * @returns true if the gesture should be prevented, false otherwise
    */
   protected shouldPreventGesture(element: TargetElement): boolean {
+    // First check if required keyboard keys are pressed
+    if (this.requiredKeys && this.requiredKeys.length > 0) {
+      if (!this.keyboardManager.areKeysPressed(this.requiredKeys)) {
+        return true; // Prevent the gesture if required keys are not pressed
+      }
```

Comment:
You can remove this line since the first thing done by `areKeysPressed` is exactly the same 
```suggestion
    // First check if required keyboard keys are pressed

      if (!this.keyboardManager.areKeysPressed(this.requiredKeys)) {
        return true; // Prevent the gesture if required keys are not pressed
      }
```